### PR TITLE
fix: Keep the consumers callback list when "registerConsumers" is called

### PR DESCRIPTION
**Pull request by janosbabik**
## Description

The RabbitMQMessageBroker now retains the list of consumers even after their initial registration. Consequently, when a reconnection event occurs, the consumers are automatically re-registered. This ensures message reception for clients.

## Motivation and Context

Currently, the RabbitMQMessageBroker resets consumer functions after their initial registration. However, in the event of a reconnection, the consumer callback functions are not re-registered due to an empty consumer list. This creates a challenge for applications as they lack awareness of reconnection events within the RabbitMQMessageBroker. Consequently, they cannot trigger the re-registration of their consumer callback functions. 

## How Has This Been Tested

- Unit Testing
- Manual Testing with the Connector

## Fixes

[https://jira.esss.lu.se/browse/SWAP-3472](https://jira.esss.lu.se/browse/SWAP-3472)

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: packages/message-broker/src/index.ts [hunk 1]
```diff
@@ -329,10 +329,7 @@ export class RabbitMQMessageBroker implements MessageBroker {
         {}
       );
```
Comment (martin-trajanovski) [inline packages/message-broker/src/index.ts:331]: Great cleanup!

File: packages/message-broker/src/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+jest.mock('amqplib');
```
Comment (martin-trajanovski) [inline packages/message-broker/src/index.spec.ts:1]: It is great that you added unit testing as well.

File: packages/message-broker/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@user-office-software/duo-message-broker",
-  "version": "1.5.1",
+  "version": "1.5.2",
```
Comment (martin-trajanovski) [inline packages/message-broker/package.json:3]: Just a small addition here: You should run `npm run lerna:prePublish` after you update package version so the package-lock.json gets updated as well.

**Review by martin-trajanovski — CHANGES_REQUESTED** [review]
It looks great. Just a small comment that package-lock.json should be updated too before it is pushed. https://github.com/UserOfficeProject/user-office-lib#publishing-new-version

File: packages/message-broker/package.json (hunk 3 above)
Comment (janosbabik) [reply packages/message-broker/package.json:3]: I did run the script and the package-lock.json was updated with the correct version number, you can see in the diff. If I miss something please let me know :)

Comment (martin-trajanovski) [reply packages/message-broker/package.json:3]: Sorry, my bad I missed it. Approved!

File: packages/message-broker/src/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    beforeEach(async () => {
+      consumer = jest.fn();
+      await broker.setup();
+    });
+
+    it('should register a consumer when "listenOn" function is called', async () => {
+      broker.listenOn(Queue.SCHEDULING_PROPOSAL, consumer);
+
+      // waiting for registerConsumer function to finish behind the scenes
+      await sleep();
```
Comment (Junjiequan) [inline packages/message-broker/src/index.spec.ts:49]: I see you have declared` const sleep = (ms = 0) => new Promise((resolve) => setTimeout(resolve, ms));`
But I don't understand how does it work without passing any argument

**Review by Junjiequan — COMMENTED** [review]
Just left one question, otherwise looks good to me

File: packages/message-broker/src/index.spec.ts (hunk 4 above)
Comment (Junjiequan) [reply packages/message-broker/src/index.spec.ts:49]: Ah, sorry, let me clarify my question..
What I wanted to ask was why do we put setTimeout with 0 seconds here, shouldn't it be 5 seconds or something?

Comment (janosbabik) [reply packages/message-broker/src/index.spec.ts:49]: Yeah, it is due to the asynchronous task in the listenOn function. The listenOn function calls an asynchronous job, but it doesn't wait for it, and it also doesn't return with it, so we can't wait for the function to finish. I didn't want to change the interface because of backward compatibility; therefore, we need to ensure that the async job in the listenOn function finishes before the expect line. With a setTimeout zero delayed timer task, we can wait for the task and test the result.
Basically, it is about controlling the execution order of tasks in the Event Loop.

Comment (Junjiequan) [reply packages/message-broker/src/index.spec.ts:49]: Thanks for the good explanation!

**Review by bolmsten — APPROVED** [review]
